cli/bpmetadata/types.go

```diff
@@ -4,92 +4,141 @@ import (
 	Info         BlueprintInfo         `json:",inline" yaml:",inline"`
 	Content      BlueprintContent      `json:",inline" yaml:",inline"`
 	Interfaces   BlueprintInterface    `json:",inline" yaml:",inline"`
 	Requirements BlueprintRequirements `json:",inline" yaml:",inline"`
 }
 
-// BlueprintInfo defines informational detail for the blueprint
 type BlueprintInfo struct {
-	Title          string
-	Source         *BlueprintRepoDetail
-	Version        string                  `json:",omitempty" yaml:",omitempty"`
-	ActuationTool  BlueprintActuationTool  `json:"actuationTool,omitempty" yaml:"actuationTool,omitempty"`
-	Description    *BlueprintDescription   `json:",omitempty" yaml:",omitempty"`
-	Icon           string                  `json:",omitempty" yaml:",omitempty"`
-	DeploymentTime BlueprintTimeEstimate   `json:"deploymentTime,omitempty" yaml:"deploymentTime,omitempty"`
-	CostEstimate   BlueprintCostEstimate   `json:",omitempty" yaml:",omitempty"`
-	CloudProducts  []BlueprintCloudProduct `json:",omitempty" yaml:",omitempty"`
-	QuotaDetails   []BlueprintQuotaDetail  `json:",omitempty" yaml:",omitempty"`
-}
-
-// BlueprintContent defines the detail for blueprint related content such as
-// related documentation, diagrams, examples etc.
-type BlueprintContent struct {
-	// Diagrams are manually entered
-	Architecture  BlueprintArchitecture  `json:"architecture,omitempty" yaml:"architecture,omitempty"`
-	Diagrams      []BlueprintDiagram     `json:",omitempty" yaml:",omitempty"`
-	Documentation []BlueprintListContent `json:",omitempty" yaml:",omitempty"`
-	SubBlueprints []BlueprintMiscContent `json:"subBlueprints,omitempty" yaml:"subBlueprints,omitempty"`
-	Examples      []BlueprintMiscContent `json:",omitempty" yaml:",omitempty"`
-}
-
-// BlueprintInterface the input and output variables for the blueprint
-type BlueprintInterface struct {
-	Variables []BlueprintVariable `json:",omitempty" yaml:",omitempty"`
-	// VariableGroups are manually entered
-	VariableGroups []BlueprintVariableGroup `json:"variableGroups,omitempty" yaml:"variableGroups,omitempty"`
-	Outputs        []BlueprintOutput        `json:",omitempty" yaml:",omitempty"`
-}
-
-// BlueprintRequirements defines the roles required and the assocaited services
-// that need to be enabled to provision blueprint resources
-type BlueprintRequirements struct {
-	Roles    []BlueprintRoles
-	Services []string
+	// Title for the solution.
+	// Autogenerated: First H1 text in readme.md
+	Title string
+	// Solution source location and source type
+	// Autogen details in BlueprintRepoDetail
+	Source *BlueprintRepoDetail
+	// Semantic version  for the packaged solution
+	// Autogenerated: From the `module_name` attribute of the `provider_meta "google"` block
+	// E.g.
+	// provider_meta "google" {
+	//  module_name = "blueprints/terraform/terraform-google-log-analysis/v0.1.5"
+	// }
+	Version string `json:",omitempty" yaml:",omitempty"`
+	// Actuation tool e.g. Terraform and its required version
```

Comment:
nit: since we are planning to use this for godocs let's make sure comments are full sentences ending with a period when applicable for consistency.

Reply:
done